pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -993,6 +980,9 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 
 	if keepCount == 0 && len(changes.ContainersToStart) == 0 {
 		changes.KillPod = true
+		// To prevent the sidecar containers to keep pod alive, we should not
```

Comment:
What scenario has KillPod true but changes.InitContainersToStart != nil?  If there is no such case, we should clear InitContainersToStart outside of this guard, and probably have the caller of computePodActions explicitly ignore InitContainersToStart?

Reply:
we keep getting `InitContainersToStart` to be not `nil` as they need to be restarted every time they crashed. So we can get into situation when we don't have containers to start, but still has sidecars to start.

This explains the logic, let me take a look if this can be optimized like you suggest